pyuvdata/cst_beam.py

```diff
@@ -204,8 +208,17 @@ def read_cst_beam(self, filename, beam_type='power', feed_pol='x',
 
         # get beam
         if self.beam_type is 'power':
-            data_col = np.where(np.array(column_names) == 'abs(v)')[0][0]
-            power_beam1 = data[:, data_col].reshape((theta_axis.size, phi_axis.size), order='F') ** 2.
+            data_col = np.where(np.logical_or(np.core.defchararray.find(np.core.defchararray.lower(np.array(column_names)), 'abs(v)') != -1,
+                                              np.core.defchararray.find(np.core.defchararray.lower(np.array(column_names)), 'abs(dir.)') != -1))[0][0]
+            power_beam1 = data[:, data_col].reshape((theta_axis.size, phi_axis.size), order='F')
+
+            mag_dbi = 'dbi' in np.core.defchararray.lower(units)[data_col]
+
+            if mag_dbi:
+                power_beam1 = 10. ** (power_beam1 / 10.)
+
+            if 'abs(v)' in column_names[data_col]:
+                power_beam1 = power_beam1 ** 2.
```

Comment:
I think the reason you're not squaring the dBi value is because the factor of 10 (vs 20) in the denominator of the exponent effectively takes care of that for you. Can you make that explicit in a comment for the future reader?

Also this logic is slightly funny in that you could hit both of these if statements, but I don't think that's actually physical -- I don't believe directivities can be express in voltages. Maybe we need some logic to catch a case that doesn't make sense followed by an if/elif that makes it clear that there are only 2 options here.

Reply:
I've added a comment and I've moved the squaring into an elif.